src/Geta.NotFoundHandler/Core/Redirects/CustomRedirectCollection.cs

```diff
@@ -79,34 +74,19 @@ public CustomRedirect Find(Uri urlNotFound)
     public void Add(CustomRedirect customRedirect)
     {
         var oldUrl = HttpUtility.UrlDecode(customRedirect.OldUrl);
-        if (_quickLookupTable.ContainsKey(oldUrl))
-        {
-            _quickLookupTable[oldUrl] = customRedirect;
-            return;
-        }
-
-        // Add to quick look up table too
-        _quickLookupTable.Add(oldUrl, customRedirect);
-
-        // clean cache
-        _redirectsZACache = null;
+        _redirectsZA[oldUrl] = customRedirect;
     }
 
     private CustomRedirect FindInternal(string url)
     {
         url = HttpUtility.UrlDecode(url) ?? string.Empty;
-        if (_quickLookupTable.TryGetValue(url, out var redirect))
+        if (_redirectsZA.TryGetValue(url, out var redirect))
         {
             return redirect;
         }
 
         // working with local copy to avoid multi-threading issues
-        var redirectsZA = _redirectsZACache;
-        if (redirectsZA == null)
-        {
-            redirectsZA = _quickLookupTable.OrderByDescending(x => x.Key, StringComparer.OrdinalIgnoreCase).ToArray();
-            _redirectsZACache = redirectsZA;
-        }
+        var redirectsZA = _redirectsZA.ToArray();
```

Comment:
This array used to be calculated and cached. With the proposed change, the array is re-created on every call to FindInternal, likely increasing memory usage and garbage collections. FindInternal is expected to be called multiple times during public Find method.